lib/idn.c

```diff
@@ -50,6 +50,67 @@
 #include "curl_memory.h"
 #include "memdebug.h"
 
+/* for macOS and iOS targets */
+#if defined(HAVE_APPLE_IDN)
+#include <unicode/uidna.h>
+
+static CURLcode mac_idn_to_ascii(const char *in, char **out)
+{
+  UErrorCode err = U_ZERO_ERROR;
+  UIDNA* idna = uidna_openUTS46(UIDNA_CHECK_BIDI, &err);
+  if(U_FAILURE(err)) {
+    return CURLE_OUT_OF_MEMORY;
+  }
+  else {
+    UIDNAInfo info = UIDNA_INFO_INITIALIZER;
+    char buffer[256] = {0};
+    int32_t inlength = -1; /* zero terminated */
+
+    int32_t len = uidna_nameToASCII_UTF8(idna, in, inlength, buffer,
+      sizeof(buffer), &info, &err);
+    uidna_close(idna);
+    if(U_FAILURE(err)) {
+      return CURLE_URL_MALFORMAT;
+    }
+    else {
+      *out = strdup(buffer);
+      if(*out)
+        return CURLE_OK;
+      else
+        return CURLE_OUT_OF_MEMORY;
+    }
+  }
+}
+
+static CURLcode mac_ascii_to_idn(const char *in, char **out)
+{
+  UErrorCode err = U_ZERO_ERROR;
+  UIDNA* idna = uidna_openUTS46(UIDNA_CHECK_BIDI, &err);
+  if(U_FAILURE(err)) {
+    return CURLE_OUT_OF_MEMORY;
+  }
+  else {
+    UIDNAInfo info = UIDNA_INFO_INITIALIZER;
+    char buffer[256] = {0};
+    int32_t inlength = -1; /* zero terminated */
+
+    int32_t len = uidna_nameToUnicodeUTF8(idna, in, inlength, buffer,
+      sizeof(buffer), &info, &err);
```

Comment:
```suggestion
    int32_t len = uidna_nameToUnicodeUTF8(idna, in, -1, /* null terminated */
      buffer, sizeof(buffer), &info, &err);
```

Reply:
Well, if you prefer, we can skip them. I like to document the meaning of the value as I don't like magic numbers that much. Or we use strlen(in) for the parameter.

Reply:
It saves having to declare a pointless variable, the "documentation" is still there. And it's not like you document all the other arguments. Finally: it's not your job to document the API function that is used.

Reply:
Okay. Change made.